### PR TITLE
fix: give iislander a more reasonable tolerance

### PR DESCRIPTION
**Pull request by WilsontheWolf**
Supersedes #248 

This makes iislander have less tolerance but still a bit more than #248.

Specifically:
- Swear cool-down is now an hour
- Swears now have a ranking.
  - Most swears are ranked at 3 which is an instant penalty
  - Some swears are listed at 1 or 2. 1 behaves like the old swear system and 2 is only one warning and then a penalty.
  - I choose the rankings based off of how I felt they should work. Please give feedback
- Swear regex now has substitutions and removes useless info
  - This will prevent people from bypassing swears with just an _ or even simple letter substitutions.
  - This is not the system we had before. It will not try finding more words based off of the other stuff.

Please give feedback.

Some other things to consider:
- Increasing penalty for multiple swears in one message
- Maybe removing some more of the advanced substitutions
- Auto deltetion on messages with higher swear levels (maybe 5 and above)
- If a user's recent penalties hits a certain number (3? 5? 10? etc?) time them out for like 24h (note this would need some changes to the core mod system and would probably be a separate pr)

### CODE REVIEW

File: src/modules/automod/message.js [hunk 1]
```diff
@@ -10,28 +10,42 @@ import { getChannel } from '../../util';
...
+        .replace(/3/g, 'e')
+        .replace(/[4@ä]/g, 'a')
+        .replace(/[5$]/g, 's')
+        .replace(/[7+]/g, 't')
+        .replace(/8/g, 'b')
+        .replace(/\(/g, 'c')
+        .replace(/ü/g, 'u')
+        .replace(/[^a-z ]/g, '')
+        .split(/(?:\b|\s)+/g);
+    const word = words.find((w) => Blacklist.swearWords.find(([swear]) => w === swear));
```
Comment (KommentatorForAll) [inline src/modules/automod/message.js:30]: should search all swears and return the highest by value

File: src/constants/Blacklist.js [hunk 2]
```diff
@@ -1,40 +1,41 @@
...
-    'nigger',
-    'porn',
-    'pussy',
-    'shit',
-    'sex',
-    'slut',
-    'tits',
-    'titties',
-    'vagina',
+    ['arse', 1],
```
Comment (KommentatorForAll) [inline src/constants/Blacklist.js:6]: use a map instead. then you can later just use .keys() for filtering which is easier to understand when looking at it than the double .filter() thing and extracting only the word from the array

File: src/modules/automod/message.js [hunk 3]
```diff
@@ -10,28 +10,42 @@ import { getChannel } from '../../util';
...
+    const [word, level] = containsSwear(message) || [];
     if (word) {
         const userDto = userDb.get(message.author.id);
         const now = Date.now();
         const cooldown = userDto.cooldown.get('swearing');
         if (now > (cooldown || 0)) {
             userDto.swearlevel = 0; // resets the swear level, if time ran out
         }
-        userDto.swearlevel += 1;
+        userDto.swearlevel += level;
```
Comment (KommentatorForAll) [inline src/modules/automod/message.js:48]: level 3 swears should be deleted as has been discussed somewhen earlier

**Review by KommentatorForAll — CHANGES_REQUESTED** [review]

File: src/modules/automod/message.js [hunk 4]
```diff
@@ -10,30 +10,56 @@ import { getChannel } from '../../util';
...
 /**
  * @param {import('discord.js').Message} message
  * @return {Promise<void>}
  */
 async function handleSwearing(message) {
-    const word = containsSwear(message);
-    if (word) {
+    if (getUserMod(message.member) >= Mod.ENFORCER) return;
+    const swears = containsSwear(message) || [];
+    console.log(swears);
```
Comment (Chicken) [inline src/modules/automod/message.js:46]: Debug in production yes

File: src/constants/Blacklist.js [hunk 5]
```diff
@@ -1,41 +1,42 @@
...
+    ['clit', 3],
+    ['cunt', 3],
+    ['cum', 3],
+    ['dick', 2],
+    ['dildo', 3],
+    ['ejaculate', 3],
+    ['motherfucker', 3],
+    ['fuck', 2],
+    ['fucker', 3],
+    ['fag', 3],
```
Comment (Chicken) [inline src/constants/Blacklist.js:23]: If we doing changes anyway, add the longer form of this to the list too.

File: src/constants/Blacklist.js [hunk 6]
```diff
@@ -1,41 +1,42 @@
...
-    'shit',
-    'sex',
-    'slut',
-    'tits',
-    'titties',
-    'vagina',
-];
+const swearWords = new Map([
+    ['arse', 1],
+    ['anus', 3],
```
Comment (KommentatorForAll) [inline src/constants/Blacklist.js:7]: 2

File: src/constants/Blacklist.js [hunk 7]
```diff
@@ -1,41 +1,42 @@
...
+    ['boob', 3],
```
Comment (KommentatorForAll) [inline src/constants/Blacklist.js:8]: 2

File: src/constants/Blacklist.js [hunk 8]
```diff
@@ -1,41 +1,42 @@
...
+    ['cum', 3],
+    ['dick', 2],
+    ['dildo', 3],
+    ['ejaculate', 3],
+    ['motherfucker', 3],
+    ['fuck', 2],
+    ['fucker', 3],
+    ['fag', 3],
+    ['gangbang', 3],
+    ['horny', 3],
```
Comment (KommentatorForAll) [inline src/constants/Blacklist.js:25]: 2

File: src/constants/Blacklist.js [hunk 9]
```diff
@@ -1,41 +1,42 @@
...
+    ['jerkoff', 3],
+    ['masturbate', 3],
+    ['nigga', 3],
+    ['nigger', 3],
+    ['porn', 3],
+    ['pussy', 3],
+    ['shit', 1],
+    ['sex', 1],
+    ['slut', 3],
+    ['tits', 3],
```
Comment (KommentatorForAll) [inline src/constants/Blacklist.js:36]: 2

File: src/constants/Blacklist.js [hunk 10]
```diff
@@ -1,41 +1,42 @@
...
+    ['titties', 3],
```
Comment (KommentatorForAll) [inline src/constants/Blacklist.js:37]: 2

**Review by KommentatorForAll — CHANGES_REQUESTED** [review]
maybe those values are a bit more balanced

**Review by KommentatorForAll — APPROVED** [review]

**Review by Chicken — APPROVED** [review]

**Pull request merged**